tests/testthat/test-compare.R

```diff
@@ -167,6 +164,37 @@ test_that("vec_proxy_compare.POSIXlt() correctly orders around DST", {
   expect_equal(vec_order(c(y, x)), c(2, 1))
 })
 
+test_that("vec_proxy_compare() flattens df-cols", {
+  df_col <- new_data_frame(list(z = 3:4, w = 4:5))
+  df <- new_data_frame(list(x = 1:2, y = df_col))
+
+  expect <- new_data_frame(list(x = 1:2, z = 3:4, w = 4:5))
+
+  expect_identical(vec_proxy_compare(df), expect)
+})
+
+test_that("vec_proxy_compare() unwraps 1 col dfs", {
+  df <- new_data_frame(list(x = 1:2))
+
+  expect_identical(vec_proxy_compare(df), 1:2)
+
+  df_col <- new_data_frame(list(y = 1:2))
+  df <- new_data_frame(list(x = df_col))
+
+  expect_identical(vec_proxy_compare(df), 1:2)
+})
+
+test_that("vec_proxy_order() works on deeply nested lists", {
+  df_col <- new_data_frame(list(z = list("b", "a", "b")))
```

Comment:
Can you use `data_frame()` please? it's easier to understand the structure without the extra list.